### DISCUSSION TITLE
Correct way to use PropertyResolver param of copy

### DISCUSSION THREAD

**Discussion by oddnugget** [donmccurdy/glTF-Transform#495] [Q&A] [answered]
I am trying to copy a mesh and and set a different material on it, but i noticed that the copy method will add children by reference, which naturally means that changing the material of the meshPrimitive will change it on both the source and copy. 

What's the correct way to implement the PropertyResolver in order to acomplish this?

**Comment by donmccurdy** [comment, marked as answer]
Mainly the resolver argument is there for internal use by `document.merge()` and `document.clone()`, but you could also use it to make a deep copy. The function takes a child (or descendant) of the original as its argument and should return the property it maps to on the target. For the simplest example:

```js
const mesh2 = document.createMesh().copy(mesh1, (ref) => ref.clone());
```

If the same texture might be used more than once you'd probably want to keep track of it and return the same clone rather than additional clones here.

**Reply by oddnugget** [reply]
Ah, makes sense. Thanks!
